File: backend/src/common/domain/models/processing/workflow_rule_compilation.py
from datetime import datetime
from typing import Any
from uuid import UUID, uuid4

from pydantic import BaseModel, ConfigDict, Field

from src.common.domain.enums.workflow_rules import WorkflowRuleCompilationStatus


class WorkflowRuleCompilation(BaseModel):
    uuid: UUID = Field(default_factory=uuid4)
    rule_id: UUID = Field(...)
    version: int = Field(...)
    kind: str = Field(...)
    status: WorkflowRuleCompilationStatus = Field(default=WorkflowRuleCompilationStatus.PENDING)
    artifact: dict[str, Any] | None = Field(default=None)
    compiled_with: dict[str, Any] | None = Field(default=None)
    error: str | None = Field(default=None)
    created_at: datetime | None = Field(default=None)
    completed_at: datetime | None = Field(default=None)

    model_config = ConfigDict(from_attributes=True, extra="ignore")

File: frontend/src/domain/entities/workflow-rule-compilation.ts
export type WorkflowRuleCompilationStatus =
  | "PENDING"
  | "COMPILING"
  | "READY"
  | "FAILED"
  | "STALE";

export interface WorkflowRuleCompilation {
  uuid: string;
  ruleId: string;
  version: number;
  kind: string;
  status: WorkflowRuleCompilationStatus;
  artifact: Record<string, unknown> | null;
  compiledWith: Record<string, unknown> | null;
  error: string | null;
  createdAt: string | null;
  completedAt: string | null;
}
